[DRAFT] Gerencia (#115)
StructCE/gitbook: execucao/ruby-on-rails/gems.md

As últimas duas configurações estão desabilitando o cop de documentação com comentário e desabilitando o cop de Strings literais imutáveis, que introduzem uma feature que não é utuilizada com frequência em Rails e que possui pouca aplicação no momento.

Além disso, algumas configurações do rubocop são em relação ao tamanho das coisas (métodos, classes, testes), nesse caso, é possível apenas alterar o limite máximo ao invés de desabilitar o cop, já que esses muita vezes são importantes. Por exemplo, para alterar o tamanho máximo dos métodos para 16 linhas, basta colocar 
```
Metrics/MethodLength:
  Max: 16
```

## Gems úteis

As *gems úteis* são gems que proporcionam funcionalidades desejáveis ou essenciais apenas em alguns casos, mas que não são necessárias em todos os projetos da Struct de *Ruby on Rails*. Como gerente de projeto, é sua responsabilidade ler a descrição das gems nessa seção e incluir e configurar apenas as gems que se apliquem ao projeto em questão.

> 

Reviewer: ```suggestion

### Mailjet
Para envios de email, pode ser utilizada a SDK (kit de desenvolvimento de software) que a Mailjet criou para o ruby.

Basta adicionar à gemfile a gem 'mailjet', e configurar os arquivos de `config/{development/production}` da maneira correta (verificar projetos passados sobre `config.action_mailer`)

```

### Figaro

O Figaro é uma gem utilizada para gerenciar credenciais de bancos de dados em um arquivo de configuração local. Recomenda-se sua utilização em projetos que possuam *um banco de dados com o SGBD \(Sistema gerenciador de banco de dados\) MySQL* ou qualquer outro SGBD que não possua um bom sistema de armazenamento de credenciais de login \(o PostgreSQL é um caso raro de SGBD que **não** precisa da gem Figaro\).

Para configurar o Figaro, siga os passos abaixo:

1. Adicione o trecho de código abaixo ao arquivo `Gemfile`:

  ```
  # Database credentials:
  gem 'figaro'
  ```

2. Execute o comando `bundle install` para instalar as gem adicionadas ao `Gemfile`.

3. Execute o comando `bundle exec figaro install` para configurar a gem.

4. Modifique o arquivo `config/database.yml` nas linhas demonstradas abaixo:

  ```
  default: &default
    # Modifique apenas as linhas abaixo e não apague ou modifique as outras linhas!
    username: <%= ENV['db_user'] %>
    password: <%= ENV['db_password'] %>
  ```

> 5. Adicione o trecho de código abaixo ao arquivo `config/application.yml`:

Reviewer: ```suggestion
5. Adicione o trecho de código abaixo ao arquivo `config/application.yml.example`:
```

  ```
  # Database credentials:
  db_user: "seu_usuario_do_DB_entre_parentesis"
  db_password: "sua_senha_do_DB_entre_parentesis"
  ```

> Após configurar a gem Figaro em seu ambiente, **avise os outros membros do projeto** que eles deveram executar os passos 2, 3 e 5 para configurar a gem em seus ambientes!

Reviewer: ```suggestion
Após configurar a gem Figaro em seu ambiente, **avise os outros membros do projeto** que eles devem criar um `config/application.yml` baseado no `.example`, colocando as próprias informações ou chaves usadas no projeto.
```

### Active Storage

Sempre que precisar trabalhar em alguma api que envolva o armazenamento de arquivos (images, documentos, audios, etc.), essa gem será sua maior aliada.

#### Configuração

A configuração inicial para essa gem é bem simples:

1. Adicione-a à Gemfile: `gem "activestorage"` e dê o bundle;

2. Execute o comando `rails active_storage:install` delegar para a gem praticamente toda a configuração inicial;

3. Execute o comando `rails db:migrate` para rodar as migrations geradas pelo comando anterior;

Para mais informações sobre como utilizar a gem, dê uma olhada no [drive da empresa](https://docs.google.com/document/d/1LfNYolHHFG0Q8C5wNrfXgJSWm1MpGV5U_hBIcOtqpB4/edit?usp=sharing).

## *Deprecated* gems

As *deprecated gems* são gems que não recebem mais manutenção e suporte por parte de um time de desenvolvedores, devendo ser retiradas dos projetos da Struct de *Ruby on Rails*. Como gerente de projeto, é sua responsabilidade substituir todas as gems dessa seção que estão presentes no seu projeto.

> ### Chrome driver helper
> 
> A gem `chromedriver-helper` parou de receber suporte em Março de 2019. Substitua ela no Gemfile pela gem `webdrivers`.
> 
> ### SASS Rails
> 
> A gem `sass-rails` parou de receber suporte em Março de 2019. Substitua ela no Gemfile pela gem `sassc-rails`.

Reviewer: São gems usadas somente em rails full stack

```suggestion
```
